Content.Shared/Chat/TypingIndicator/SharedTypingIndicatorSystem.cs

```diff
@@ -90,4 +92,23 @@ private void SetTypingIndicatorEnabled(EntityUid uid, bool isEnabled, Appearance
 
         _appearance.SetData(uid, TypingIndicatorVisuals.IsTyping, isEnabled, appearance);
     }
+
+    /// <summary>
+    /// DeltaV: Adds an override to the TypingIndicator visuals
+    /// </summary>
+    /// <param name="protoId">The TypingIndicator to use in place of default or clothing indicators. Clears overrides when null.</param>
+    private void SetTypingOverride(EntityUid uid, ProtoId<TypingIndicatorPrototype>? protoId, AppearanceComponent? appearance = null)
+    {
+        if (!Resolve(uid, ref appearance, false))
+            return;
+
+        if (protoId != null)
+        {
+            _appearance.SetData(uid, TypingIndicatorVisuals.OverrideProto, protoId, appearance);
+        }
+        else
+        {
+            _appearance.RemoveData(uid, TypingIndicatorVisuals.OverrideProto);
+        }
```

Comment:
add an optional networked protoid to TypingIndicatorComponent instead of abusing appearance